src/core/tests/test_evented.py

```diff
@@ -0,0 +1,148 @@
+import unittest
+from unittest.mock import create_autospec, call, patch, PropertyMock, Mock
+from toga.evented import Evented, Event, UndefinedEventRaised
+
+
+class EventedTests(unittest.TestCase):
+    class ABaseWidget(Evented):
+        on_simple_event1 = Event('Simple Event 1')
+        on_simple_event2 = Event('Simple Event 2')
```

Comment:
from an API perspective, I'd say the "on_" prefix is redundant in the declaration. The handler is "on_simple_event1", but the event that is raised id "simple_event1".

Reply:
Do we want the object attributes themselves to have the `on_` prefix? If we do, I'm not sure allowing us to have it one way during definition and another during use is worth the extra code complexity that would be needed to achieve that...

Reply:
Depends what exactly you mean by "internal". I'd expect that the end users will be setting `button.on_click = some_method` and passing in `Button(on_click=some_method)`, and I'd expect to be able to confirm that there's an event handler by checking `if interface.on_click`, but if that's backed by a different property name internally, that would be fine by me.

Reply:
Well if you expect it to be called `on_click` in all the places you mention, then its much simpler code-wise to also call it `on_click` when declaring it as opposed to `click`. Doing the latter will probably require some meta-programming gymnastics I'd rather avoid.

Since identify event via string when raising them, we could easily use a slightly different name there (E.g. without the `on_` prefix), but at that point I think it may be better to be consistent.

But I'm beginning to think I'm failing to understand what you are trying to say, so let me try to ask what I need to know in a bit more detail. Currently we use the `on_click` style event name in all the following places:

1. When defining the event as part of the Widget class definition
2. When setting even handler via the constructor
3. When setting even handler via an object property
4. When querying the event handler via an object property
5. When notifying the backend object about a new callback via the `set_event_handler` method (Event name is passed as a string)
6. When the backend asks the interface to call the callbacks via a call to `raise_event` (Event name is passed as a string).

In which of those cases do you think we should be using the `click` style name? Do you think doing that is worth having code that may be harder to understand?